crates/shared/src/order_validation.rs

```diff
@@ -513,6 +533,23 @@ impl OrderValidating for OrderValidator {
             OrderCreationAppData::Full { full } => validate(full)?,
         };
 
+        // convert the user-specified hooks into interactions.
+        if !app_data.backend.hooks.is_empty() {
+            // custom interactions are disabled
+            if !self.enable_custom_interactions {
+                return Err(ValidationError::Partial(
+                    PartialValidationError::UnsupportedCustomInteraction,
+                ));
+            }
+            // custom interactions are not allowed for partially fillable orders
+            if order.partially_fillable {
+                return Err(ValidationError::Partial(
+                    PartialValidationError::UnsupportedCustomInteraction,
+                ));
+            }
```

Comment:
I think with the introduction of hooks this restriction could actually be lifted eventually.
AFAIK the only problem here is that was unclear in the context of partial fills whether a custom interaction should be executed on every fill or only on the first or last fill.
With hooks we could theoretically support arbitrarily complex rules traders could use.
That being said, I think it would be too much hassle for a probably very niche scenario.

Reply:
We had it in the MVP spec to not support this for partially fillable orders. Agreed that now that we have "hooks" we can add support eventually (by specifying when the hook should be executed - first part, when its done, every time, etc.).